### PR TITLE
Write filtered phenotypes to a file flag

### PR DESCRIPTION
**Pull request by nathanielparke**
This PR allows a command line user to chose to write all the filtered phenotypes to a directory of their choosing. Useful for debugging data issues.

### CODE REVIEW

**Comment by devin-petersohn** [comment]
pinging @fnothaft @heuermh to approve Jenkins build test.

File: gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala [hunk 1]
```diff
@@ -68,5 +68,52 @@ class LogisticValidationRegressionSuite extends GnocchiFunSuite {
...
+    val statistics = Map("weights" -> weights)
+    val assoc = Association(fakeVariant, "", 0.0, statistics)
+
+    // Array[(String, Double)] where String is sampleid and Double is predicted value
+    val predictionResult = AdditiveLogisticEvaluation.predictSite(sampleObservations, assoc, 0.3)
+
+    // Assert that the predictions result in the same as the actual phenotype (on dummy set)
+    for (i <- predictionResult.indices) {
+      if (predictionResult(i) != expectedResults(i)) {
+        print("Error --> ")
```
Comment (heuermh) [inline gnocchi-core/src/test/scala/net/fnothaft/gnocchi/association/LogisticValidationRegressionSuite.scala]: these debugging statements shouldn't be left in, make assertions and let the test framework report errors

File: gnocchi-cli/src/main/scala/net/fnothaft/gnocchi/cli/RegressPhenotypes.scala [hunk 2]
```diff
@@ -95,6 +95,10 @@ class RegressPhenotypesArgs extends Args4jBase {
 
   @Args4jOption(required = false, name = "-oneTwo", usage = "If cases are 1 and controls 2 instead of 0 and 1")
   var oneTwo = false
+
+  @Args4jOption(required = false, name = "-writeMissingPheno", usage = "Write missing phenotypes to the location supplied.")
```
Comment (heuermh) [inline gnocchi-cli/src/main/scala/net/fnothaft/gnocchi/cli/RegressPhenotypes.scala]: for better or worse ADAM cli uses `-single_dash_snake_case` for command line arguments

**Review by heuermh — CHANGES_REQUESTED** [review]

**Comment by heuermh** [comment]
Jenkins, add to whitelist

**Pull request closed without merging**